### PR TITLE
pre-23.11 release checks/changes

### PR DESCRIPTION
**Pull request by adwk67**
## Description

pre-23.11 release checks/changes. See [Nuclino](https://app.nuclino.com/Stackable/Engineering/Release-2311-d892cfac-d9de-4210-8d27-c0290b3441f4) page.

**N.B. DO NOT MERGE UNTIL AFTER 23.11 RELEASE**
- paths need to be reverted/adapted

## Review Checklist
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added (or not applicable)
- [ ] Documentation added (or not applicable)
- [ ] Changelog updated (or not applicable)
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)

### CODE REVIEW

File: demos/demos-v2.yaml [hunk 1]
```diff
@@ -9,8 +9,8 @@ demos:
     manifests:
       - plainYaml: https://raw.githubusercontent.com/stackabletech/stackablectl/main/demos/airflow-scheduled-job/01-airflow-spark-clusterrole.yaml
       - plainYaml: https://raw.githubusercontent.com/stackabletech/stackablectl/main/demos/airflow-scheduled-job/02-airflow-spark-clusterrolebinding.yaml
-      - plainYaml: https://raw.githubusercontent.com/stackabletech/stackablectl/main/demos/airflow-scheduled-job/03-enable-and-run-spark-dag.yaml
-      - plainYaml: https://raw.githubusercontent.com/stackabletech/stackablectl/main/demos/airflow-scheduled-job/04-enable-and-run-date-dag.yaml
+      - plainYaml: demos/airflow-scheduled-job/03-enable-and-run-spark-dag.yaml
```
Comment (sbernauer) [inline demos/demos-v2.yaml:12]: Marker for all the URLs to be changed back before merging

File: stacks/airflow/airflow.yaml [hunk 2]
```diff
@@ -35,7 +33,7 @@ spec:
           AIRFLOW__CORE__DAGS_FOLDER: "/dags"
           AIRFLOW_CONN_KUBERNETES_IN_CLUSTER: "kubernetes://?__extra__=%7B%22extra__kubernetes__in_cluster%22%3A+true%2C+%22extra__kubernetes__kube_config%22%3A+%22%22%2C+%22extra__kubernetes__kube_config_path%22%3A+%22%22%2C+%22extra__kubernetes__namespace%22%3A+%22%22%7D"
         replicas: 1
-  workers:
+  celeryExecutors:
```
Comment (sbernauer) [inline stacks/airflow/airflow.yaml:36]: We could also switch to use the k8s executor - or create a follow-up issue

**Review by sbernauer — COMMENTED** [review]
Thanks for working on this! LGTM in general
While you are at it, you could also bump third party stuff such as postgres, minio, redis, opensearch in https://github.com/stackabletech/stackablectl/tree/main/stacks/_templates. It's just because testing the stuff is the most effort - so no need to do so in case you already completed your tests ;)

File: demos/data-lakehouse-iceberg-trino-spark/create-spark-ingestion-job.yaml [hunk 3]
```diff
@@ -41,8 +41,9 @@ data:
       name: spark-ingest-into-lakehouse
     spec:
       version: "1.0"
-      sparkImage: docker.stackable.tech/demos/pyspark-k8s-with-kafka-and-iceberg:3.3.0-stackable23.4
-      sparkImagePullPolicy: Always
+      sparkImage:
+        custom: docker.stackable.tech/demos/pyspark-k8s-with-kafka-and-iceberg:3.3.0-stackable0.2.0
+        productVersion: 3.3.0
```
[the remark this answers is not in the document]
Comment (adwk67) [reply demos/data-lakehouse-iceberg-trino-spark/create-spark-ingestion-job.yaml:46]: Thanks - that would be great if you could look at the lakehouse demo!

File: stacks/airflow/airflow.yaml (hunk 2 above)
Comment (adwk67) [reply stacks/airflow/airflow.yaml]: Thanks for this suggestion - it improves the job clean-up . Done: [ab6d6e6](https://github.com/stackabletech/stackablectl/pull/290/commits/ab6d6e68651c4c3978b4f3a98295ec9debe17f63)

Comment (sbernauer) [reply stacks/airflow/airflow.yaml]: Awesome, many thanks!

File: demos/data-lakehouse-iceberg-trino-spark/create-spark-ingestion-job.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (sbernauer) [reply demos/data-lakehouse-iceberg-trino-spark/create-spark-ingestion-job.yaml:46]: Sure, will! If it's ok I would do it separately afterwards

File: stacks/stacks-v2.yaml [hunk 4]
```diff
@@ -66,7 +66,7 @@ stacks:
     manifests:
       - helmChart: https://raw.githubusercontent.com/stackabletech/stackablectl/main/stacks/_templates/postgresql-airflow.yaml
       - helmChart: https://raw.githubusercontent.com/stackabletech/stackablectl/main/stacks/_templates/redis-airflow.yaml
-      - plainYaml: https://raw.githubusercontent.com/stackabletech/stackablectl/main/stacks/airflow/airflow.yaml
+      - plainYaml: stacks/airflow/airflow.yaml
```
[the remark this answers is not in the document]
Comment (maltesander) [reply stacks/stacks-v2.yaml:69]: there are many of these, i would just grep / replace demo/... stacks/.. before merging.

[the remark this answers is not in the document]
Comment (adwk67) [reply stacks/stacks-v2.yaml]: Done.

**Review by maltesander — APPROVED** [review]
LGTM!

**Pull request merged**